server/webapp/dbmodels.py

```diff
@@ -660,32 +660,51 @@ def restore(self, progset, program_list):
                 program = self._program_to_dict(program)
                 update_or_create_program(self.project.id, self.id, program_name, program, True)
 
-    def create_programs_from_list(self, programs):
+    def recreate_programs_from_list(self, programs, progset_id):
         from optima.utils import saves
 
         prog_shorts = []
-        for program in programs:
-            kwargs = {}
-            for field in ['name', 'short', 'category', 'targetpops', 'pars', 'costcov', 'criteria']:
-                kwargs[field] = program[field]
+        desired_shorts = set([program.get('short', program.get('short_name', '')) for program in programs])
+        existing_programs = db.session.query(ProgramsDb).filter_by(progset_id=progset_id)
 
-            # Kind of a hack but sometimes we receive short ans sometimes short_name
-            if 'short_name' in program and program['short_name'] is not None:
-                kwargs['short'] = program['short_name']
```

Comment:
see in the old code here. short is used, but then if short_name exists, short is replaced by short_name